hq/test/logic/SnykDisplayTest.scala

```diff
@@ -0,0 +1,291 @@
+      projects.getFailedAttempt.failures.head.friendlyMessage shouldBe "Could not read Snyk response (some nice error)"
+    }
+
+    "fail to find project id list (not nice) - fails" in {
+      val projects = SnykDisplay.getProjectIdList(SnykDisplayTest.mockBadResponseWithoutMessage)
+      projects.isFailedAttempt shouldBe true
+    }
+    "fail to find project id list (not nice) - has message" in {
+      val projects = SnykDisplay.getProjectIdList(SnykDisplayTest.mockBadResponseWithoutMessage)
+      projects.getFailedAttempt.failures.head.friendlyMessage shouldBe """Could not read Snyk response ({"toughluck": "no use"})"""
+    }
+  }
+
+  "find empty vulnerability list" in {
+    val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockGoodAndNotVulnerableResponse))
+    projects.value().head.ok shouldBe true
+  }
+
+  "find non-empty vulnerability list" - {
+    "find ok" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockGoodButVulnerableResponse))
+      projects.value().head.ok shouldBe false
+    }
+
+    "find high count" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockGoodButVulnerableResponse))
+      projects.value().head.high shouldBe 1
+    }
+
+    "find medium count" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockGoodButVulnerableResponse))
+      projects.value().head.medium shouldBe 0
+    }
+
+    "find low count" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockGoodButVulnerableResponse))
+      projects.value().head.low shouldBe 0
+    }
+  }
+
+  "fail to find vulnerability list (nice)" - {
+    "fail to find vulnerability list (nice) - fails" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockBadResponseWithMessage))
+      projects.isFailedAttempt() shouldBe true
+    }
+    "fail to find vulnerability list (nice) - has message" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockBadResponseWithMessage))
+      projects.getFailedAttempt.failures.head.friendlyMessage shouldBe "Could not read Snyk response (some nice error)"
+    }
+  }
+
+  "fail to find vulnerability list (not nice)" - {
+    "fail to find vulnerability list (not nice) - fails" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockBadResponseWithoutMessage))
+      projects.isFailedAttempt() shouldBe true
+    }
+    "fail to find vulnerability list (not nice) - has message" in {
+      val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockBadResponseWithoutMessage))
+      projects.getFailedAttempt.failures.head.friendlyMessage shouldBe """Could not read Snyk response ({"toughluck": "no use"})"""
+    }
+  }
+
+  "label organisation" - {
+    "label organisation - first id" in {
```

Comment:
The test message should say what property we're checking so that when it fails the developer sees something helpful.

Reply:
It does ... it checks the first id.

Reply:
The failure message for this test will be precisely:

    label organisation label organisation - first id

The duplication of label organisation is unhelpful and there's not much info beyond that.

You know what that means because you wrote it, but the person dealing with a failing test will not see the implementation, only the above string. What do you think they will be able to infer from it?

What property of the id are you testing? In this case you are checking that the resulting id is equal to `id0`. What's the significance of `id0`? It's not in the `goodProjects` test data so where is it coming from and why?

The message doesn't need to be an essay, but it should help guide the understanding of someone that has to fix the project. I like to think of tests as compiled comments, if you were going to write an inline comment on the relevant line of code, what would it say?